plugins/obs-filters/compressor-filter.c

```diff
@@ -28,6 +28,9 @@
 /* clang-format off */
 
 #define S_RATIO                         "ratio"
+#if defined(_M_ARM64)
+	#undef S_THRESHOLD
```

Comment:
Why is this being undefined here?

Reply:
The `#undef S_THRESHOLD` is necessary to avoid a macro redefinition warning.
This is because `windows.h`, which includes `winbase.h`, defines `S_THRESHOLD` with a value of 1. Our source file requires `S_THRESHOLD` to be defined as `"threshold"`.

Reply:
Looks like a namespace conflict (here `S_`), the reasonable change would be to change all `compressor-filter.c` `S_` macros to something like `FILTER_S_` in a **separate** PR.

Reply:
The majority of the plugins use `S_` as the prefix to define setting names. Changing the macro name makes the history dirty. IMHO, `#undef` is reasonable.

There are various ways in which the setting names are written among the 1st-party plugins. Unfortunately, the majority write string literals directly without macros. The 2nd place is `S_`.

Besides this PR, I want to suggest the style should be unified among the plugins. In my opinion, `SETTING_`, `OPT_`, or `PROP_` (no one uses this though) would be descriptive rather than `S_`.

Plugins directly writing string literals
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/coreaudio-encoder/encoder.cpp#L1375-L1386
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/image-source/color-source.c#L22-L24
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/image-source/image-source.c#L104-L107
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/linux-alsa/alsa-input.c#L190-L201
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/linux-capture/xcomposite-input.c#L855-L863
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/linux-capture/xshm-input.c#L275-L283
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/linux-jack/jack-input.c#L61-L62
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/linux-pipewire/camera-portal.c#L1307-L1330
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/linux-pipewire/screencast-portal.c#L609-L611
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/linux-pulseaudio/pulse-input.c#L552
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/linux-v4l2/v4l2-input.c#L1154-L1163
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/mac-avcapture/plugin-properties.m#L321-L326
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/mac-videotoolbox/encoder.c#L823-L833
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-ffmpeg/obs-ffmpeg-av1.c#L69-73
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-ffmpeg/obs-ffmpeg-hls-mux.c#L143
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-ffmpeg/obs-ffmpeg-mpegts.c#L976-L977
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-ffmpeg/obs-nvenc.c#L647-L657
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/crop-filter.c#L76-L82
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/eq-filter.c#L47-L49
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/hdr-tonemap-filter.c#L76-L86
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/scroll-filter.c#L79-L89
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/sharpness-filter.c#L26
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-libfdk/obs-libfdk.c#L96-L99
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-outputs/flv-mux.c#L139
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-outputs/flv-output.c#L509
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-outputs/mp4-output.c#L263-L274
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-qsv11/obs-qsv11.c#L262-344
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-transitions/transition-stinger.c:#L70-L72
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-x264/obs-x264.c#L402-L416
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/text-freetype2/text-freetype2.c#L341-358
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/win-capture/duplicator-monitor-capture.c
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/win-capture/monitor-capture.c
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/win-capture/window-capture.c
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/win-dshow/win-dshow-encoder.cpp

Image slideshow uses `static const char *` with the prefix `S_`. (Probably, this is Lain's preference, isn't this?)
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/image-source/obs-slideshow-mk2.c#L19-L33

Plugins using macros with the prefix `S_`
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/compressor-filter.c#L30-L35
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/expander-filter.c#L30-L37
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/gain-filter.c#L12
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/gpu-delay.c#L5
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/limiter-filter.c#L28-L29
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/noise-gate-filter.c#L14-L18
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/noise-suppress-filter.c#L43-L50
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/nvidia-greenscreen-filter.c#L25-L30
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/scale-filter.c#L11-L30
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-text/gdiplus/obs-text.cpp#L46-L80
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-transitions/transition-fade-to-color.c#L3-L7
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-transitions/transition-luma-wipe.c#L7-L9
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-transitions/transition-slide.c#L5
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-transitions/transition-swipe.c#L17-L18
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/vlc-video/vlc-video-source.c#L15-L25

Plugins using macros with a prefix `SETTING_`
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/async-delay-filter.c#L17
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/chroma-key-filter.c#L8-L17
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/color-correction-filter.c#L23-L32
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/color-grade-filter.c#L9-L12
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/color-key-filter.c#L9-12
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/color-key-filter.c
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/luma-key-filter.c
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-filters/mask-filter.c
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/win-capture/game-capture.c

Plugins using macros without prefix
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/decklink/const.h#L1-L17
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/win-dshow/win-dshow.cpp#L34-L50
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/win-wasapi/win-wasapi.cpp#L27-L30

Plugins using macros with a prefix `OPT_`
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/obs-outputs/rtmp-stream.c#L26-L34
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/win-wasapi/win-wasapi.cpp#L27-L30

Only AJA plugin has a structure.
- https://github.com/obsproject/obs-studio/blob/ff2fa24dc25519475687041ba6af829e9ba41335/plugins/aja/aja-ui-props.hpp

Reply:
> Changing the macro name makes the history dirty.

Undefining macros that are not ours is not a good idea either.

Reply:
Poking @paulh-aja for insight in to AJA's efforts/status on ARM support.

Reply:
To clarify, the larger conversation here is whether or not we should explore a more comprehensive change that renames our macros to avoid the conflicts with generic defines such as `S_THRESHOLD`. The smaller scoped conversation is "are we okay with undefining this macro to get this to build?" The secondary question is "does `WIN32_LEAN_AND_MEAN` solve this?"

Reply:
The `winbase.h` header file defines `S_THRESHOLD` with a value of `1`, and this definition cannot be excluded by the `WIN32_LEAN_AND_MEAN` directive.
So the available options are to `#undef S_THRESHOLD` or Renaming the OBS macro which would affect the project's history, as mentioned by @norihiro